packages/ai-chat-ui/src/browser/chat-response-renderer/toolcall-part-renderer.tsx

```diff
@@ -75,7 +75,11 @@ export class ToolCallPartRenderer implements ChatResponsePartRenderer<ToolCallCh
         let responseContent = response.result;
         try {
             if (response.result) {
-                responseContent = JSON.stringify(JSON.parse(response.result), undefined, 2);
+                let resultObject = response.result;
```

Comment:
Should we do this conversion in the model implementation? Is this a corner case?